### ISSUE TITLE
Missing deps when building @fmt ?

### ISSUE THREAD

**Issue by hhugo** [ocaml/dune#2464]
related PRs: https://github.com/ocaml/dune/pull/1824

**Comment by hhugo** [comment]
It seems that the `@fmt` alias doesn't depend on `.ocamlformat` and `.ocamlformat-ignore` files

**Comment by nojb** [comment]
This is probably related to #2404

**Comment by nojb** [comment]
I tried to reproduce but it seems to work correctly in my machine. @hhugo, could you provide detailed reproduction instructions? Thanks!

**Comment by rgrinberg** [comment]
In particular it would be nice to see the `$ dune rules -m @fmt`. It will give us an immediate answer if the file is considered in the rules.

**Comment by hhugo** [comment]
I got the issue with js_of_ocaml 
```
$ dune --version
1.11.0
$ git clone git@github.com:ocsigen/js_of_ocaml.git jsoo-clone
$ cd jsoo-clone
$ opam install ocamlformat.0.10
$ dune build @fmt
```

**Comment by nojb** [comment]
Thanks, I was able to reproduce and am investigating.

**Comment by nojb** [comment]
It seems to be a problem registering the dependency of `.ocamlformat` files when they are not in the workspace root (which is why it worked in my simple test).

**Comment by nojb** [comment]
The regression is due to no longer using the `--name FILE` flag in the invocation of `ocamlformat`. It turns out that some `ocamlformat` options will be looked up in `.ocamformat{,-ignore}` files in the parent directories of `FILE` (this is all that is said in the `man` page of `ocamlformat`).

Before #2404 `FILE` was the filename in the source directory and the `.ocamlformat{,-ignore}` was looked up there. After #2404 this option `--name` is no longer passed but the `.ocamlformat{,-ignore}` are not copied into `_build` so they are not found by `ocamlformat`.

Using `--name` is not completely obvious now because as far as I can see we do not have easy access to the source filename for `%{input-file}`, so I will try instead to copy the `.ocamlformat{,-ignore}` files into `_build`.

**Comment by nojb** [comment]
@hhugo fix in #2471. Thanks for the report!
